TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODSlotsPlugin.cs

```diff
@@ -183,201 +255,177 @@ public override void OnConnection(VMEODClient client)
+                            betIsValid = true;
+                    }
                 }
-                else
-                    client.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.InsufficientFunds));
-            });
+            }
+
+            if (betIsValid == true)
+            {
+                // roll the slots rng and get result, send to UI and send to SlotResultHandler() to determine amount of CurrentWinnings, CurrentWinnings = 0 is a loss
+                var thisGameBetAndResults = RollNewGame(betAmount);
+                CurrentWinnings = SlotResultHandler(thisGameBetAndResults);
+
+                // attempt to debit player the bet amount
+                var VM = client.vm;
+
+                VM.GlobalLink.PerformTransaction(VM, false, client.Avatar.PersistID, Server.Object.PersistID, Convert.ToInt32(betAmount),
+
+                (bool success, int transferAmount, uint uid1, uint budget1, uint uid2, uint budget2) =>
+                {
+                //TODO: Make this part of global link
+                VM.SendCommand(new VMNetAsyncResponseCmd(0, new VMTransferFundsState
+                    {
+                        Responded = true,
+                        Success = success,
+                        TransferAmount = transferAmount,
+                        UID1 = uid1,
+                        Budget1 = budget1,
+                        UID2 = uid2,
+                        Budget2 = budget2
+                    }));
+                    if (success)
+                    {
+                    // update the balance of the machine
+                    MachineBalance = Convert.ToInt16(budget2);
+
+                        if (CurrentWinnings > 0)
+                        {
+                        // dispatch event to tell object the winning amount
+                        UserClient.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.SetWinAmount, new short[1] { Convert.ToInt16(CurrentWinnings) }));
+                        // dispatch event to tell the object that the user has won
+                        UserClient.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.ExecuteWin, new short[1] { betAmount }));
+                        }
+                        else
+                        {
+                        // dispatch event to tell object that the user has lost
+                        UserClient.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.ExecuteLoss, new short[1] { betAmount }));
+                        }
+                        // send spins results to UI to simulate gameplay
+                        UserClient.Send("slots_spin", new Byte[] { (byte)SlotsStops[thisGameBetAndResults[1]], (byte)SlotsStops[thisGameBetAndResults[2]],
+                        (byte)SlotsStops[thisGameBetAndResults[3]] });
+                    }
+                    else
+                    {
+                        CurrentWinnings = 0; // forfeit any winnings
```

Comment:
Someone could bet money they don't have to set the CurrentWinnings, then try to get the winnings before the game realizes they cannot be paid for the bet.

Store the winnings in a local variable outside of the transaction, then ONLY set CurrentWinnings to that value when the transaction succeeds.